### DISCUSSION TITLE
Working towards translating templates

### DISCUSSION THREAD

**Discussion by perdedora** [vichan-devel/vichan#535] [Ideas]
Last month or so, I began to try to understand how to generate .po file from twig strings since the last update of .po files lacks several new strings. The script `i18n_extract.php` only works well when all of the templates are cached and not so much ideal for a development environment due to constantly cleaning the cache or just using with debug or auto reload on. 
Then I started searching for alternatives to generate .po files from the strings in the twig itself, POEdit Free is not suitable because the Twig parser is behind the PRO license and all of the parsers that I tried failed to extract all strings. So after a little searching I made a script to compile all templates, put them in a folder then scan the project folder with POEdit. It works very well and I was able to translating all of the strings.

I'm bringing this here to explore more alternatives to this problem, suggestions and etc.

Link to the script: https://gist.github.com/perdedora/eb4b5891e5d2499416bc9e21ddb96932
Just save in the `tools` folder and run from the cli.

**Comment by Zankaria** [comment]
So, coming from android development, we could copy the stategy used by android apps.
With android, string internationalization is centralized in a few `strings.xml` files, similar to `$config['error']`.
We could make a separate `trans.php` file in a similar vain: containing a single giant array.

**Comment by chanadmins** [comment]
I gave up on enabling translation because changing names of things breaks random javascripts and the translations themselves are messy and unfinished

**Reply by perdedora** [reply]
Using the script I made I was able to translate everything. Translating javascript was a little tricky because the library we used was abandoned and needed a little patch to work with modern php.

If you choose the same route as me:
Search paths for POEdit:
https://github.com/perdedora/nicochan/blob/master/inc/locale/pt_BR/LC_MESSAGES/javascript.po#L25-L33
https://github.com/perdedora/nicochan/blob/master/inc/locale/pt_BR/LC_MESSAGES/tinyboard.po#L25-L39
